File: src/test/java/com/d20charactersheet/framework/boc/service/DnDv35RuleServiceMoneyTest.java
package com.d20charactersheet.framework.boc.service;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import com.d20charactersheet.framework.boc.model.Character;
import com.d20charactersheet.framework.boc.model.Money;

public class DnDv35RuleServiceMoneyTest extends DnDv35RuleServiceTestCase {

  @Test
  public void testGetMoney() throws Exception {
    final Money money = new Money();
    money.setPlatinum(1);
    money.setGold(2);
    money.setSilver(3);
    money.setCopper(4);

    final Character character = createCharacter(fighter, 1);
    character.setMoney(money);
    final float gold = ruleService.getGold(character);
    assertEquals(12.34f, gold, 0.0f);
  }
}

File: src/test/java/com/d20charactersheet/framework/boc/service/DnDv35RuleServiceMoneyTest.kt
package com.d20charactersheet.framework.boc.service

import com.d20charactersheet.framework.boc.model.Money
import org.junit.Assert
import org.junit.jupiter.api.Test

class DnDv35RuleServiceMoneyTest : DnDv35RuleServiceTestCase() {

    @Test
    fun testGetMoney() {
        val money = Money()
        money.platinum = 1
        money.gold = 2
        money.silver = 3
        money.copper = 4
        val character = createCharacter(fighter, 1)
        character.money = money
        val gold = ruleService.getGold(character)
        Assert.assertEquals(12.34f, gold, 0.0f)
    }
}